string[] paths
string[] system_path
string source_node_name
---
